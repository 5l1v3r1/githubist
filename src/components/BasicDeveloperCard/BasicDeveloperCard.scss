@import '../../theme/theme';

.basicDeveloperCard {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.avatar {
  margin-bottom: map-get($spacers, normal);
}

.nameAndUsername {
  text-decoration: none;

  &:hover {
    color: $black-color;
    text-decoration: underline;
  }
}

.name {
  color: $black-color;
  font-weight: 600;
}

.username {
  margin-left: map-get($spacers, 2x-small);
  color: $body-color;
}

.info {
  overflow: hidden;
  width: 100%;
  margin: map-get($spacers, 2x-small) 0 0 0;
  font-size: map-get($font-sizes, small);
  text-align: center;
  text-overflow: ellipsis;
  white-space: nowrap;
}
