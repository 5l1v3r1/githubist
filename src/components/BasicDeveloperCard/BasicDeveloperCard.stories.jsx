// @flow

import React from 'react';
import { storiesOf } from '@storybook/react';
import { MemoryRouter } from 'react-router-dom';

import BasicDeveloperCard from './BasicDeveloperCard';

storiesOf('BasicDeveloperCard', module)
  .add('default', () => (
    <MemoryRouter>
      <BasicDeveloperCard
        profilePicture="https://via.placeholder.com/150"
        name="Alpcan Aydın"
        username="alpcanaydin"
        info="@Atolye15"
      />
    </MemoryRouter>
  ))
  .add('with long info', () => (
    <MemoryRouter>
      <div style={{ width: '250px' }}>
        <BasicDeveloperCard
          profilePicture="https://via.placeholder.com/150"
          name="Alpcan Aydın"
          username="alpcanaydin"
          info="This is a very long info text that needs to be truncated"
        />
      </div>
    </MemoryRouter>
  ))
  .add('without info', () => (
    <MemoryRouter>
      <BasicDeveloperCard
        profilePicture="https://via.placeholder.com/150"
        name="Alpcan Aydın"
        username="alpcanaydin"
      />
    </MemoryRouter>
  ));
