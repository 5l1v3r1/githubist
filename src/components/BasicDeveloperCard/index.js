// @flow

import BasicDeveloperCard from './BasicDeveloperCard';

export default BasicDeveloperCard;
