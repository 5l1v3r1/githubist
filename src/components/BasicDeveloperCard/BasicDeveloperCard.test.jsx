// @flow

import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';

import BasicDeveloperCard from './BasicDeveloperCard';

it('renders correctly', () => {
  const wrapper = mount(
    <MemoryRouter keyLength={0}>
      <BasicDeveloperCard
        profilePicture="https://via.placeholder.com/150"
        name="Alpcan Aydın"
        username="alpcanaydin"
        info="@Atolye15"
      />
    </MemoryRouter>,
  );

  expect(wrapper).toMatchSnapshot();
});

it('renders correctly without info', () => {
  const wrapper = mount(
    <MemoryRouter keyLength={0}>
      <BasicDeveloperCard
        profilePicture="https://via.placeholder.com/150"
        name="Alpcan Aydın"
        username="alpcanaydin"
      />
    </MemoryRouter>,
  );

  expect(wrapper).toMatchSnapshot();
});
