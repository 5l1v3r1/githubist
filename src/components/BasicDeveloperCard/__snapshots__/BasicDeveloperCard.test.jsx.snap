// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <BasicDeveloperCard
      info="@Atolye15"
      name="Alpcan Aydın"
      profilePicture="https://via.placeholder.com/150"
      username="alpcanaydin"
    >
      <div>
        <Link
          replace={false}
          to="/alpcanaydin"
        >
          <a
            href="/alpcanaydin"
            onClick={[Function]}
          >
            <Avatar
              alt="Alpcan Aydın"
              size="default"
              url="https://via.placeholder.com/150"
            >
              <img
                alt="Alpcan Aydın"
                className=""
                src="https://via.placeholder.com/150"
              />
            </Avatar>
          </a>
        </Link>
        <Link
          replace={false}
          to="/alpcanaydin"
        >
          <a
            href="/alpcanaydin"
            onClick={[Function]}
          >
            <span>
              Alpcan Aydın
            </span>
            <span>
              alpcanaydin
            </span>
          </a>
        </Link>
        <p>
          @Atolye15
        </p>
      </div>
    </BasicDeveloperCard>
  </Router>
</MemoryRouter>
`;

exports[`renders correctly without info 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <BasicDeveloperCard
      name="Alpcan Aydın"
      profilePicture="https://via.placeholder.com/150"
      username="alpcanaydin"
    >
      <div>
        <Link
          replace={false}
          to="/alpcanaydin"
        >
          <a
            href="/alpcanaydin"
            onClick={[Function]}
          >
            <Avatar
              alt="Alpcan Aydın"
              size="default"
              url="https://via.placeholder.com/150"
            >
              <img
                alt="Alpcan Aydın"
                className=""
                src="https://via.placeholder.com/150"
              />
            </Avatar>
          </a>
        </Link>
        <Link
          replace={false}
          to="/alpcanaydin"
        >
          <a
            href="/alpcanaydin"
            onClick={[Function]}
          >
            <span>
              Alpcan Aydın
            </span>
            <span>
              alpcanaydin
            </span>
          </a>
        </Link>
      </div>
    </BasicDeveloperCard>
  </Router>
</MemoryRouter>
`;
